app/news-page/stores/edit-post.store.ts

```diff
@@ -0,0 +1,70 @@
+import { FormState } from 'formstate';
+import { action, observable } from 'mobx';
+
+import { formStateToJS, InputFieldState, TextAreaFieldState } from '../../common/utils/form-helpers';
+import { FormValidators } from '../../common/utils/form-validators';
+import { NewsApi } from '../api/news.api';
+import { Post } from '../models/post';
+
+export class EditPostStore {
+    result: Promise<Post | undefined>;
+    private close!: (saved?: Post) => void;
+
+    @observable
+    serverError = '';
+
+    @observable
+    post: Post;
+
+    form: ReturnType<typeof EditPostStore.createForm>;
+
+    constructor(
+        private newsApi: NewsApi,
+        post: Post,
+    ) {
+        this.post = post;
+        this.form = EditPostStore.createForm(post);
+        this.result = new Promise(r => this.close = r)
+    }
+
+    private static createForm(post: Post) {
```

Comment:
Any benefits of static method in this case?

Reply:
Just to suppress webstorm's `Method can be static` warning